reference/types/promise.md

```diff
@@ -1,5 +1,14 @@
 # Promise
 
-A write-once container which sets its value in the future. Similar to [Future][type-future].
+A write-once container which sets its value in the future. Similar to [Future][type-future]. It is a proxy for a value which may not be known at the time the promise is created and allows for asynchronous execution<sup>1</sup>. Promises can be in one of three states:
+- Pending: the initial state before the promise has been completed
+- Fulfilled or Resolved: the success state after the operation is successfully completed
+- Rejected: the error state signalling that the operation has failed
 
 [type-future]: ./future.md
+
+---
+
+[1] Promise, MDN Web Docs. (2020). https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise (accessed June 3, 2020).
```

Comment:
Reference seems lang specific (unsure if that is of concern or not for reference docs).

Would something more lang generic be preferable for the case where non-JS peeps want to look up about promises? (e.g. https://en.wikipedia.org/wiki/Futures_and_promises)

Reply:
I would indeed prefer a language-agnostic reference here. We can fix this later though. I'm fine with merging this as is.

Reply:
Hey guys, thanks for the suggestions. Yes, I agree that something language-agnostic would be better, or otherwise (or in addition) a short sentence for each major language. I'm not experienced writing promises/futures in languages other than JS, so am less confident in writing about it. Also the line (between a promise and a future) seems to be quite blurry - i.e. it looks like C++ has both.

Anyway, I added 'in JavaScript' to, hopefully, make it less confusing for people. Maybe someone else can extend this to include other langauges :)

Reply:
@sgedye Maybe you could link to https://en.wikipedia.org/wiki/Futures_and_promises? That sounds like a good introduction. You can then remove the JavaScript addition.

Reply:
Updated.